Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,252 @@
+﻿using Data;
+using Data.Models.ProductTables;
+using Backend.Services;
+using Lib.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using Lib.Storage;
+using Lib.Storage.Services;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using Lib;
+
+namespace Backend.Controllers
+{
+    [Route("v1/products")]
+    [ApiController]
+    [Authorize]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly IStorage storage;
+
+        private readonly Jwt jwt;
+
+
+        public ProductController(Db context, ImageService imageService, IStorage storage, Jwt jwt)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+            this.jwt = jwt;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(CreateProductInput newProduct)
+        {
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.Id == newProduct.ShopId && s.OwnerId == uid);
+           
+            if (shop == null) return Forbid();
+
+            Product product = new Product();
```

Comment:
make constructor for product. DO NOT ASSIGN value outside of constructor when creating object